paper/paper.md

```diff
@@ -0,0 +1,167 @@
+figures. The Jupyter notebook is ideal for educators because it allows them to
+create assignments which include instructions along with code or Markdown cells,
+in which students can provide solutions to exercises. Students can, for example,
+write code both to compute and visualize a particular result. Because the
+notebook is interactive, students can iterate on a coding problem without
+needing to switch back and forth between a command line and a text editor, and
+they can rapidly see results alongside the code which produced them.
+
+Instructors in many fields have begun using the Jupyter notebook as a teaching
+platform. The notebook has appeared in over 70 classes
+[@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics,
+mechanical engineering, data science, chemical engineering, and bioinformatics.
+Software Carpentry, an organization that aims to teach graduate students basic
+computational skills, has also adopted the notebook for some of its lessons
+[@Wilson2014].
+
+Despite its appearance in many classrooms, before to the existence of nbgrader,
+the notebook was rarely used on a large scale for *graded* assignments. Instead,
+it was often used either for ungraded in-class exercises, or in classes small
+enough that notebooks could be graded by hand. This is because there are several
+challenges to using the notebook for graded assignments at scale. First, for
+large classes, it is not feasible for an instructor to manually grade the code
+that students write: there must be a way to autograde the assignments. However,
+a Jupyter notebook is not a typical script that can be run and may contain
+multiple parts of a problem within the same notebook; thus, autograding a
+notebook is less straightforward than it is for a traditional script. Second,
+for many courses, programming is a means to an end: understanding concepts in a
+specific domain. Specifically, instructors may want students to provide both
+code and written free-responses interpreting the results of that code.
+Instructors thus need to be able to rely on autograding for the coding parts of
+an assignment, but also be able to manually grade the written responses in the
+surrounding context of the student's code. Third, the process of distributing
+assignments to students and later collecting them can be tedious, even more so
+with Jupyter notebooks because there is a separate interface for accessing them
+beyond the standard system file browser. This often leads to confusion on the
+part of students about how to open notebooks after downloading them, and where
+to find the notebooks in order to submit them.
+
+nbgrader streamlines the repetitive tasks found in course management and
+grading, and its flexibility allows greater communication between instructor and
+student. Overall, nbgrader improves the learning experience for both instructors
+and students, enabling them to focus on content and building understanding.
+
```

Comment:
It sounds like nbgrader is done, complete. Should there be a sentence stating that, like all good open source projects, development continues?